### PR TITLE
Update outdated documentation for BitmapFrontEnd.

### PR DESCRIPTION
**Pull request by glintfishGH**
- Changed arguments to start with a lowercase letter.
- Fixed documentation ported over from flash(?), where the documented @:params would not align to function arguments.

### CODE REVIEW

**Comment by Geokureli** [comment]
Oh I just merged this, thinking it was you
https://github.com/HaxeFlixel/flixel/pull/3312

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 1]
```diff
@@ -160,10 +160,10 @@ class BitmapFrontEnd
 	}
 
 	/**
-	 * Caches specified FlxGraphic object.
+	 * Caches a specified FlxGraphic object.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:163]:
```suggestion
	 * Caches the specified graphic.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 2]
```diff
@@ -160,10 +160,10 @@ class BitmapFrontEnd
...
 	 *
 	 * @param	graphic	FlxGraphic to store in the cache.
-	 * @return	cached FlxGraphic object.
+	 * @return	Cached FlxGraphic object.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:166]:
```suggestion
	 * @return	The cached graphic.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 3]
```diff
@@ -123,18 +123,18 @@ class BitmapFrontEnd
 	}
 
 	/**
-	 * Generates a new BitmapData object (a colored rectangle) and caches it.
+	 * Creates a new FlxGraphic object (a colored rectangle) and caches it.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:126]:
```suggestion
	 * Creates a new graphic of a colored rectangle and caches it.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 4]
```diff
@@ -114,7 +114,7 @@ class BitmapFrontEnd
 	/**
 	 * Check the local bitmap cache to see if a bitmap with this key has been loaded already.
 	 *
-	 * @param	Key		The string key identifying the bitmap.
+	 * @param	Key	The key identifying the bitmap.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:117]: Doc params should:
- start with lowercase letters
- have 3 spaces before the arg name and at least 2 after
- have all arg descriptions start on the same column
Other function doc rules:
- `@return` should have 2 spaces after, so the description aligns with the params
- Types should be wrapped in `backticks`
- Avoid types in args with default values (`unique:Bool = false` can be `unique = false`)

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 5]
```diff
@@ -123,18 +123,18 @@ class BitmapFrontEnd
...
-	 * @param	Color	What color the rectangle should be (0xAARRGGBB)
-	 * @param	Unique	Ensures that the bitmap data uses a new slot in the cache.
-	 * @param	Key		Force the cache to use a specific Key to index the bitmap.
-	 * @return	The BitmapData we just created.
+	 * @param	width	How wide the rectangle should be.
+	 * @param	height	How high the rectangle should be.
+	 * @param	color	What color the rectangle should be (0xAARRGGBB).
+	 * @param	unique	Ensures that the bitmap data uses a new slot in the cache.
+	 * @param	key		Force the cache to use a specific Key to index the bitmap.
+	 * @return	The FlxGraphic we just created.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:133]: here's an example of the formatting, and lets just call it a "graphic" not a "`FlxGraphic` object"
```suggestion
	 * @param   width   How wide the rectangle should be.
	 * @param   height   How high the rectangle should be.
	 * @param   color     What color the rectangle should be (0xAARRGGBB).
	 * @param   unique  Ensures that the bitmap data uses a new slot in the cache.
	 * @param   key       Force the cache to use a specific Key to index the bitmap.
	 * @return  The created graphic.
```

All of these suggestions can and should be committed using the dropdown below each. Please update the other methods

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 6]
```diff
@@ -123,18 +123,18 @@ class BitmapFrontEnd
...
 	 */
-	public function create(Width:Int, Height:Int, Color:FlxColor, Unique:Bool = false, ?Key:String):FlxGraphic
+	public function create(width:Int, height:Int, color:FlxColor, unique:Bool = false, ?key:String):FlxGraphic
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:135]:
```suggestion
	public function create(width:Int, height:Int, color:FlxColor, unique = false, ?key:String):FlxGraphic
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 7]
```diff
@@ -160,10 +160,10 @@ class BitmapFrontEnd
 	}
 
 	/**
-	 * Caches specified FlxGraphic object.
+	 * Caches a specified FlxGraphic object.
 	 *
 	 * @param	graphic	FlxGraphic to store in the cache.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:165]:
```suggestion
	 * @param   graphic  The graphic to cache.
```

**Review by Geokureli — CHANGES_REQUESTED** [review]

**Comment by glintfishGH** [comment]
For future reference, where are the doc rules? didn't even know they existed

**Comment by Geokureli** [comment]
lol they aren't formally listed anywhere. I should make that

File: flixel/system/frontEnds/BitmapFrontEnd.hx (hunk 4 above)
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:117]: - make `key` lower case
- put 2 spaces after `key` and 3 before (currently a tab)

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 8]
```diff
@@ -160,10 +160,10 @@ class BitmapFrontEnd
 	}
 
 	/**
-	 * Caches specified FlxGraphic object.
+	 * Caches the specified graphic.
 	 *
-	 * @param	graphic	FlxGraphic to store in the cache.
-	 * @return	cached FlxGraphic object.
+	 * @param   graphic  The graphic to cache.
+	 * @return	The cached graphic.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:166]:
```suggestion
	 * @return  The cached graphic.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 9]
```diff
@@ -172,20 +172,21 @@ class BitmapFrontEnd
 	}
 
 	/**
-	 * Gets FlxGraphic object from this storage by specified key.
-	 * @param	key	Key for FlxGraphic object (its name)
-	 * @return	FlxGraphic with the key name, or null if there is no such object
+	 * Gets an FlxGraphic object from this storage from its key.
+	 * 
+	 * @param	key	The FlxGraphics key (or name).
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:177]:
```suggestion
	 * @param	key	 The FlxGraphics key (or name).
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 10]
```diff
@@ -172,20 +172,21 @@ class BitmapFrontEnd
...
 	}
 
 	/**
-	 * Gets key from bitmap cache for specified BitmapData
+	 * Gets a key from a cached BitmapData.
 	 *
-	 * @param	bmd	BitmapData to find in cache
-	 * @return	BitmapData's key or null if there isn't such BitmapData in cache
+	 * @param  bmd BitmapData to find in the cache.
+	 * @return The BitmapData's key or null if there isn't such BitmapData in cache.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:189]:
```suggestion
	 * @param   bmd  BitmapData to find in the cache.
	 * @return   The BitmapData's key or null if there isn't such BitmapData in cache.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 11]
```diff
@@ -201,7 +202,7 @@ class BitmapFrontEnd
 	/**
 	 * Helper method for getting cache key for FlxGraphic objects created from the class.
 	 *
-	 * @param	source	BitmapData source class.
+	 * @param	source BitmapData source class.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:205]:
```suggestion
	 * @param   source  BitmapData source class.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 12]
```diff
@@ -201,7 +202,7 @@ class BitmapFrontEnd
...
 	 * @return	Full name for provided class.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:206]:
```suggestion
	 * @return  Full name for provided class.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 13]
```diff
@@ -230,10 +231,10 @@ class BitmapFrontEnd
 	}
 
 	/**
-	 * Gets unique key for bitmap cache
+	 * Gets unique key for bitmap cache.
 	 *
-	 * @param	baseKey	key's prefix
-	 * @return	unique key
+	 * @param	baseKey	key's prefix.
+	 * @return	unique key.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:237]:
```suggestion
	 * @param   baseKey  key's prefix.
	 * @return  unique key.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 14]
```diff
@@ -260,11 +261,11 @@ class BitmapFrontEnd
...
-	 * @param	baseKey			Beginning of the key. Usually it is the key for original spritesheet graphics (like "assets/tile.png")
-	 * @param	frameSize		the size of tile in spritesheet
-	 * @param	frameSpacing	offsets between tiles in offsets
-	 * @param	region			region of image to use as spritesheet graphics source
-	 * @return	Generated key for spritesheet with inserted spaces between tiles
+	 * @param	baseKey			Beginning of the key. Usually it is the key for original spritesheet graphics (like "assets/tile.png").
+	 * @param	frameSize		The size of tile in spritesheet.
+	 * @param	frameSpacing	Offsets between tiles in offsets.
+	 * @param	region			Region of image to use as spritesheet graphics source.
+	 * @return	Generated key for spritesheet with inserted spaces between tiles.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:268]:
```suggestion
	 * @param   baseKey       Beginning of the key. Usually it is the key for original spritesheet graphics (like "assets/tile.png").
	 * @param   frameSize     The size of tile in spritesheet.
	 * @param   frameSpacing  Offsets between tiles in offsets.
	 * @param   region        Region of image to use as spritesheet graphics source.
	 * @return  Generated key for spritesheet with inserted spaces between tiles.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 15]
```diff
@@ -287,7 +288,7 @@ class BitmapFrontEnd
 
 	/**
 	 * Totally removes specified FlxGraphic object.
-	 * @param   graphic  object you want to remove and destroy.
+	 * @param graphic The object you want to remove and destroy.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:291]:
```suggestion
	 * @param   graphic  The object you want to remove and destroy.
```

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 16]
```diff
@@ -302,7 +303,7 @@ class BitmapFrontEnd
 
 	/**
 	 * Totally removes FlxGraphic object with specified key.
-	 * @param	key	the key for cached FlxGraphic object.
+	 * @param key Key of the cached FlxGraphic object.
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:306]:
```suggestion
	 * @param   key  Key of the cached graphic.
```

Calling it a "FlxGraphic object" is wordy and requires backticks, lets just say "graphic"

**Review by Geokureli — CHANGES_REQUESTED** [review]

**Comment by glintfishGH** [comment]
whoops, forgot that @ also tags people.

**Comment by moxie-coder** [comment]
> whoops, forgot that @ also tags people.

lol, didn’t know someone had that GitHub name

**Comment by Geokureli** [comment]
Thanks!

**Pull request merged**